hipe4ml/training_utils.py

```diff
@@ -57,8 +57,9 @@ def evaluate_hyperparams(data, training_columns, reg_params, hyp_params, metrics
 
     """
     model = xgb.XGBClassifier()
-    hyp_params = cast_model_params(model, hyp_params)
-    params = {**reg_params, **hyp_params}
+    hyp_copy = hyp_params.copy()
+    hyp_copy = cast_model_params(model, hyp_copy)
+    params = {**reg_params, **hyp_copy}
```

Comment:
Is a copy needed? If yes maybe a [deep copy](https://docs.python.org/2/library/copy.html) is what you want